packages/CodingStandard/src/Fixer/DependencyInjection/InjectToConstructorInjectionFixer.php

```diff
@@ -0,0 +1,239 @@
+            [
+                new CodeSample(
+'<?php
+class SomeClass
+{
+    /**
+     * @inject
+     * @var stdClass
+     */
+    public $property;
+}'),
+            ]
+        );
+    }
+
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return $tokens->isAllTokenKindsFound([T_CLASS, T_DOC_COMMENT, T_PUBLIC, T_VARIABLE]);
+    }
+
+    public function fix(SplFileInfo $file, Tokens $tokens): void
+    {
+        // 1. find annotation @inject
+        foreach ($tokens as $index => $token) {
+            if (! $token->isGivenKind(T_DOC_COMMENT)) {
+                continue;
+            }
+
+            $doc = new DocBlock($token->getContent());
+            $injectAnnotations = $doc->getAnnotationsOfType('inject');
+            if (! count($injectAnnotations)) {
+                continue;
+            }
+
+            // 1. remove it
+            foreach ($injectAnnotations as $injectAnnotation) {
+                $injectAnnotation->remove();
+            }
+
+            $token->setContent($doc->getContent());
+
+            // 2. make public property private
+            for ($i = $index;; ++$i) {
+                $token = $tokens[$i];
+                if ($token->isGivenKind(T_PUBLIC)) {
+                    $token->override([T_PRIVATE, 'private']);
+                    break;
+                }
+            }
+
+            // 3. add dependency to constructor
+            $propertyName = '';
+            for ($i = $index;; ++$i) {
+                $token = $tokens[$i];
+                if ($token->isGivenKind(T_VARIABLE)) {
+                    $propertyName = ltrim($token->getContent(), '$');
+                    break;
+                }
+            }
+            $varAnnotation = $doc->getAnnotationsOfType('var')[0];
+            $propertyType = $varAnnotation->getTypes()[0];
+
+            // detect constructor
+            $constructMethodPosition = null;
+            for ($i = $index; $i < count($tokens); ++$i) {
+                $token = $tokens[$i];
+                if ($token->isGivenKind(T_FUNCTION)) {
+                    $namePosition = $tokens->getNextNonWhitespace($i);
+                    $methodNameToken = $tokens[$namePosition];
+                    if ($methodNameToken->getContent() === '__construct') {
+                        $constructMethodPosition = $i;
+                        break;
+                    }
+                }
+            }
+
+            // A. has a constructor?
+            if (is_int($constructMethodPosition)) { // "function" token
+                $this->addPropertyToConstructor($tokens, $propertyType, $propertyName, $constructMethodPosition);
+            } else {
+                // B. doesn't have a constructor?
+                $this->addConstructorMethod($tokens, $propertyType, $propertyName);
+            }
+
+            // run again with new tokens; @todo: can this be done any better to notice new __construct method added
+            // by these tokens?
```

Comment:
Sorry to hear and see you and the project got bitten pretty rough on the changes between 2.3 and 2.4. We tried to move as careful as we could to a better code base on which we can keep building towards 3.0. Sadly this caused rework and rethinking for others, however we thought in the end it would be for the best as not to have such a big change between 2.0 and 3.0 as there was between 1.0 and 2.0. If you've the time please keep track on what we are preparing for the upcoming releases, it might ease the transition a bit and we can take your considerations and concerns in mind.